zapisy/apps/enrollment/records/models/records.py

```diff
@@ -331,6 +331,52 @@ def free_spots_by_role(cls, group: Group) -> Dict[str, int]:
         ret['-'] = group.limit - len(all_enrolled_students)
         return ret
 
+    @classmethod
+    def taken_spots_by_role(cls, groups: List[Group]) -> Dict[int, Dict[Group, int]]:
```

Comment:
```suggestion
    def taken_spots_by_role(cls, groups: List[Group]) -> Dict[int, Dict[str, int]]:
```
Chyba jednak ten typ zostaje bez zmian, tj. kluczami wewnętrznych słowników są nazwy "ról" (bądź puste napisy związane z osobami bez ról).